From: Michael Biebl <biebl@debian.org>
Date: Tue, 18 Mar 2014 10:49:13 +0100
Subject: Force online state with unmanaged devices

If NM has an active unmanaged device it will forcefully set the online
state to CONNECTED_GLOBAL. In that case show a wired connection icon
instead of an offline icon.

Closes: #471191
---
 src/applet.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/applet.c b/src/applet.c
index 8ebac75..b105907 100644
--- a/src/applet.c
+++ b/src/applet.c
@@ -2717,6 +2717,10 @@ applet_update_icon (gpointer user_data)
 		applet_get_device_icon_for_state (applet, &pixbuf, &icon_name_free, &dev_tip_free);
 		icon_name = icon_name_free;
 		dev_tip = dev_tip_free;
+		if (!pixbuf && state == NM_STATE_CONNECTED_GLOBAL) {
+			icon_name = g_strdup ("nm-device-wired");
+			pixbuf = g_object_ref (nma_icon_check_and_load (icon_name, applet));
+		}
 		break;
 	}
 
